### ISSUE TITLE
tests hang infinitely with 0.25.5 when dependency is missing, fail as expected on 0.25.4

### ISSUE THREAD

**Issue by capaj** [vitest-dev/vitest#2451]
### Describe the bug

I have a codebase with a missing dependency where 0.25.4 fails as expected like this:
![image](https://user-images.githubusercontent.com/1305378/206133239-21424314-c55d-459c-ae8e-f5fe4247a436.png)

but 0.25.5 just hangs like this:
![image](https://user-images.githubusercontent.com/1305378/206133561-29d303d2-bdd9-4938-bf6f-8c86b541fd68.png)

### Reproduction

I can't share the repo, but if you are not able to reproduce on your own, I could prepare a repo for reproduction.
I think this happens when you try to load a dependency which is no specified in package.json, but it is in your node_modules.

### System Info

```shell
System:
    OS: Linux 5.15 Ubuntu 22.04.1 LTS 22.04.1 LTS (Jammy Jellyfish)
    CPU: (24) x64 AMD Ryzen 9 5900X 12-Core Processor
    Memory: 25.73 GB / 62.74 GB
    Container: Yes
    Shell: 5.8.1 - /usr/bin/zsh
  Binaries:
    Node: 18.12.1 - ~/.nvm/versions/node/v18.12.1/bin/node
    Yarn: 1.22.17 - ~/.yarn/bin/yarn
    npm: 8.19.2 - ~/.nvm/versions/node/v18.12.1/bin/npm
  Browsers:
    Brave Browser: 108.1.46.134
    Chrome: 108.0.5359.94
    Firefox: 107.0.1
  npmPackages:
    vite: ^3.2.4 => 3.2.5 
    vitest: 0.25.5 => 0.25.5
```

### Used Package Manager

pnpm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by sheremet-va** [comment]
> I can't share the repo, but if you are not able to reproduce on your own, I could prepare a repo for reproduction.
I think this happens when you try to load a dependency which is no specified in package.json, but it is in your node_modules.

Reproduction is always required. Please, don't make maintainers do extra work.

**Comment by capaj** [comment]
> Reproduction is always required. Please, don't make maintainers do extra work.

I would not dream of it, I just thought some maintainer could be able to realize some recent commit could cause this without even trying to reproduce.

I tried to reproduce here: https://stackblitz.com/edit/vitest-dev-vitest-nva74f?file=vitest.config.ts
but failed. It runs ok on stackblitz.
For me it fails locally and on github CI too.

Can I keep the issue opened? Maybe someone else will be able to reproduce this issue. If I close this it is unlikely anyone would ever look at it even if they got a similar problem.

**Comment by capaj** [comment]
this is fixed in 0.25.6, thanks @sheremet-va 

![image](https://user-images.githubusercontent.com/1305378/206454093-d96a2cab-aa0b-48d4-83e0-f16856a68a99.png)
